src/main/java/felix/task/Task.java

```diff
@@ -0,0 +1,70 @@
+package felix.task;
+
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+public abstract class Task {
+    protected static final DateTimeFormatter inputFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm");
+    protected static final DateTimeFormatter outputFormatter = DateTimeFormatter.ofPattern("d MMM yyyy HHmm");
+    private final String description;
+    private boolean isDone;
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    /**
+     * Returns a String representing the task's status as done.
+     */
+    protected String getStatusIcon() {
+        return (isDone ? "X" : " ");
+    }
+
+    public void markAsDone() {
+        this.isDone = true;
+    }
+
+    public void unmarkDone() {
+        this.isDone = false;
+    }
+
+    protected String getDescription() {
+        return this.description;
+    }
+
+    public abstract String toFileString();
+
+    /**
+     * Returns true if the specified keyword is a substring of the task description and false otherwise.
+     */
+    public boolean hasKeyword(String keyword) {
+        return this.description.contains(keyword);
+    }
+
+    /**
+     * Creates a task from a given line of a file within the hard disk.
+     * If necessary, the task will be marked as done.
+     *
+     * @return task represented by file line.
+     */
+    public static Task createTaskFromFileLine(String line) {
+        String[] words = line.split(" \\| ");
+        Task task;
+        if (words[0].equals("T")) {
+            task = new ToDo(words[2]);
+        } else if (words[0].equals("D")) {
+            task = new Deadline(words[2], words[3]);
+        } else {
+            task = new Event(words[2], words[3], words[4]);
+        }
+        if (words[1].equals("X")) {
+            task.markAsDone();
+        }
+        return task;
+    }
```

Comment:
I like how the `Task` class handles creation of the task objects so task types are hidden from `Storage`! Overlooked this myself. A nit may be a switch statement will make this look cleaner.

Reply:
After some thought, I think this can be improved further by moving this to a separate `TaskFactory` class so as to not have `Task` "know" its subclasses.

Reply:
Thank you for the feedback! Just to clarify, what is the purpose of the TaskFactory class?

Reply:
Instead of having a `createTask` method inside the `Task` base class itself, which somewhat violates OOP since `Task` "knows" its subclasses, I made another class `TaskFactory` which is responsible for returning the `Task` of the relevant runtime type.